src/main/java/org/jenkinsci/plugins/ewm/DiskInfoProvider.java

```diff
@@ -1,45 +1,52 @@
 package org.jenkinsci.plugins.ewm;
 
+import hudson.DescriptorExtensionList;
 import hudson.ExtensionList;
 import hudson.ExtensionPoint;
 import hudson.model.AbstractDescribableImpl;
 import jenkins.model.Jenkins;
 import org.kohsuke.accmod.Restricted;
 import org.kohsuke.accmod.restrictions.NoExternalUse;
 
-import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
 
 /**
+ * Contains additional information about a {@link org.jenkinsci.plugins.ewm.definitions.Disk} entry.
+ *
  * @author Alexandru Somai
  */
 @Restricted(NoExternalUse.class)
 public abstract class DiskInfoProvider extends AbstractDescribableImpl<DiskInfoProvider> implements ExtensionPoint {
 
-    @CheckForNull
-    private final Double writeSpeed;
+    private final double readSpeed;
+    private final double writeSpeed;
```

Comment:
Does Jenkins care about the unit, e.g. Mbps, KB/s, etc? I think not. In theory the user could be told by his admin that disk A is fast, disk B is slower than A. So the user could get away by entering 10 for disk A, and 1 for disk B, regardless of actual disk speed. What matters is that a higher number means a faster disk.

In the case of Double vs another data type (e.g. Long), it is a matter of convenience. It is not easy to visually convert Gbps and Mbps in "kilobytes per second" units, and it is not easy to read/visually compare big numbers (e.g. 1000000 - how many zeros?). But I don't think we're imposing a unit.

Implemented as "Double", we're giving the user the chance to specify disk speed in any unit of their choice, Gbps, Mbps, KB/s, etc. And the user can specify things like disk A is 2 (Gbps) and disk B is 1.2 (Gbps). So do we want to tell the user "use any unit you want but be consistent otherwise comparisons won't work.", or do we want to impose a unit of speed? I prefer that we don't impose a unit. Ten years from now, maybe even Mbps will give numbers that are unread-ably long.